File: packages/java/parser/src/test/java/com/vaadin/hilla/parser/plugins/transfertypes/push/PushTypeEndpoint.java
package com.vaadin.hilla.parser.plugins.transfertypes.push;

import com.vaadin.hilla.EndpointSubscription;
import reactor.core.publisher.Flux;

@Endpoint
public class PushTypeEndpoint {

    public Flux<String> getMessageFlux() {
        return Flux.just();
    }

    public EndpointSubscription<String> getSubscription() {
        return new EndpointSubscription<>();
    }

}

File: packages/java/typescript-generator/src/test/resources/com/vaadin/hilla/parser/plugins/transfertypes/push/snapshots/PushTypeEndpoint.ts
import type { Subscription } from '@vaadin/hilla-frontend';
import client from './connect-client.default.js';

export function getMessageFlux(): Subscription<string | undefined> {
  return client.subscribe('PushTypeEndpoint', 'getMessageFlux', {});
}

export function getSubscription(): Subscription<string | undefined> {
  return client.subscribe('PushTypeEndpoint', 'getSubscription', {});
}
